### PR TITLE
Tidy config/hostconfig structures

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

OCD moment :) Tidying the config and hostconfig structures into alphabetical order, grouping by platform agnostic, and platform specific fields.

### CODE REVIEW

File: image/fixtures/pre1.9/expected_computed_id [hunk 1]
```diff
@@ -1 +1 @@
-sha256:fd6ebfedda8ea140a9380767e15bd32c6e899303cfe34bc4580c931f2f816f89
+sha256:731d62ca192955d38edd4333c89aad021002b1e570daa6bb9f8f06b500c76a4d
```
Comment (thaJeztah) [inline image/fixtures/pre1.9/expected_computed_id]: Unrelated change?

Comment (thaJeztah) [inline image/fixtures/pre1.9/expected_computed_id]: (Not sure, just checking :smile:)

**Comment by thaJeztah** [comment]
I like it, but let's see what others think :smile:

File: image/fixtures/pre1.9/expected_computed_id (hunk 1 above)
Comment (jhowardmsft) [inline image/fixtures/pre1.9/expected_computed_id]: Nope, is related the hash will change as the ordering of elements has changed.

Comment (thaJeztah) [inline image/fixtures/pre1.9/expected_computed_id]: Ah, makes sense (I stated suspecting this with my second comment) lol

**Comment by jhowardmsft** [comment]
Rebased after LXC removal.

**Comment by thaJeztah** [comment]
ping @calavera @tiborvass you were working on these structs; looks like an easy one (lol, but that's just me :smile:)

**Comment by LK4D4** [comment]
LGTM

**Comment by calavera** [comment]
LGTM

**Comment by aaronlehmann** [comment]
The problem with this cleanup is that it will change computed image IDs. For example, if you pull an image with 1.9.0, then upgrade to a version with this commit and pull again, you'll get a different image ID and have to repull all the data. The old image will become a dangling image.

The test fixtures that are changed here were meant to prevent this from happening accidentally.

I would suggest reverting this commit to avoid the cache bust.

**Comment by calavera** [comment]
agreed with @aaronlehmann. Reverting this commit in a new PR.

**Comment by calavera** [comment]
https://github.com/docker/docker/pull/17838

**Comment by jhowardmsft** [comment]
Oh that's a bummer :(

**Comment by jhowardmsft** [comment]
@aaronlehmann Just to confirm, tidying the hostconfig structure won't affect the hash I believe. That's the messier one anyway.

**Comment by aaronlehmann** [comment]
Yes, I believe the hostconfig change is fine.
